### PR TITLE
Yzhbankov/gh 97

### LINKED ISSUE #97: Add Middlewares
Add Middlewares

add 'standard' middlewares

https://github.com/zorko-io/zorko-platform/blob/e12e13d08ad9f4fe13b584f159314f9a15ed0766/apps/web-portal/lib/WebPortalExpressApp.mjs#L30

```javascript

  initRoutes() {
    // TODO Add Middlewares
    //  add 'standard' middlewares
    //  labels: enhancement
    this.#http.use(
      '/api/v1',
      RestApiV1.route({
```

d46db70280ddaabf3edb5678dcf1cc05536160c2

### PR DESCRIPTION
**Pull request by yzhbankov**
#close gh-97

### CODE REVIEW

File: apps/web-portal/lib/WebPortalExpressApp.mjs [hunk 1]
```diff
@@ -3,6 +3,7 @@ import express from 'express'
 import {makeRunner} from '@zorko-io/util-use-case'
 import {MockLogger} from '@zorko-io/util-logger'
 import * as RestApiV1 from './rest-api-v1'
+import middlewares from './middlewares.js'
```
Comment (nesterone) [inline apps/web-portal/lib/WebPortalExpressApp.mjs]: better `import {cors, json, urlencoded} from './middlewares'`

File: apps/web-portal/lib/middlewares.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import cors from 'cors'
+import bodyParser from 'body-parser'
+
+export default {
+  json: bodyParser.json({
```
Comment (nesterone) [inline apps/web-portal/lib/middlewares.js]: export them as high-order functions, so 

`export const json = (config = { limit : 1024 * 1024}) => bodyParser.json({..})`

same for `cors` and `urlencoded`

File: apps/web-portal/lib/middlewares.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+            code: 'BROKEN_JSON',
+            message: 'Please, verify your json',
+          },
+        })
+        throw new Error('BROKEN_JSON')
+      }
+    },
+  }),
+  urlencoded: bodyParser.urlencoded({extended: true}),
+  cors: cors({origin: '*'}), // Allow any origin because NOT USE cookies and basic auth
```
Comment (nesterone) [inline apps/web-portal/lib/middlewares.js]: Add a TODO here to configure cors according to auth,

also, make an origin configurable with app config

File: apps/web-portal/lib/middlewares.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        JSON.parse(buf)
+      } catch (e) {
+        res.send({
+          status: 0,
+          error: {
+            code: 'BROKEN_JSON',
+            message: 'Please, verify your json',
+          },
+        })
+        throw new Error('BROKEN_JSON')
```
Comment (nesterone) [inline apps/web-portal/lib/middlewares.js]: Create an error `InvalidJsonError` as a sub-class from UserFacesError, and put it here, near middleware

File: apps/web-portal/lib/middlewares.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import cors from 'cors'
+import bodyParser from 'body-parser'
+
+export default {
+  json: bodyParser.json({
+    limit: 1024 * 1024,
```
Comment (nesterone) [inline apps/web-portal/lib/middlewares.js]: make limit configurable

**Review by nesterone — CHANGES_REQUESTED** [review]

File: apps/web-portal/lib/WebPortalExpressApp.mjs (hunk 1 above)
Comment (yzhbankov) [reply apps/web-portal/lib/WebPortalExpressApp.mjs]: done

File: apps/web-portal/lib/middlewares.js (hunk 2 above)
Comment (yzhbankov) [reply apps/web-portal/lib/middlewares.js]: done

File: apps/web-portal/lib/middlewares.js (hunk 3 above)
Comment (yzhbankov) [reply apps/web-portal/lib/middlewares.js]: done

File: apps/web-portal/lib/middlewares.js (hunk 4 above)
Comment (yzhbankov) [reply apps/web-portal/lib/middlewares.js]: done

File: apps/web-portal/lib/middlewares.js (hunk 5 above)
Comment (yzhbankov) [reply apps/web-portal/lib/middlewares.js]: I will move the configuration to the app config after the gh-111 will be merged

Comment (nesterone) [reply apps/web-portal/lib/middlewares.js]: extracted to separate task https://github.com/zorko-io/zorko-platform/issues/161

**Comment by nesterone** [comment]
@yzhbankov, plz resolve conflicts

**Review by nesterone — APPROVED** [review]

**Pull request merged**